packages/py-moose-lib/moose_lib/streaming/streaming_function_runner.py

```
@@ -48,6 +51,50 @@
+# Context variable to track function name without mutating globals
+_function_context: contextvars.ContextVar[Optional[str]] = contextvars.ContextVar(
+    "function_context", default=None
+)
+
+# Store original print for restoration
+_original_print = builtins.print
+
+
+# Structured print wrapper that respects kwargs and uses contextvars
+def _structured_print(
+    *args: Any,
```

Ruff (0.14.13)

Dynamically typed expressions (typing.Any) are disallowed in `*args`

(ANN401)